Sources/Testing/Support/FileHandle.swift

```diff
@@ -371,7 +373,7 @@ extension FileHandle {
 extension FileHandle {
   /// Is this file handle a TTY or PTY?
   var isTTY: Bool {
-#if SWT_TARGET_OS_APPLE || os(Linux)
+#if SWT_TARGET_OS_APPLE || os(Linux) || os(WASI)
```

Comment:
Can we assume WASI doesn't support TTYs, or is it possible for this to come back `true`?

Reply:
wasi-libc provides `isatty` implementation (not a stub!), so it can be `true` https://github.com/WebAssembly/wasi-libc/blob/main/libc-bottom-half/sources/isatty.c

Reply:
You can read from a WASI-compatible module's stdout, so it's a matter of the consumer of that stdout whether they support terminal emulation or not. There are definitely in-browser terminal emulation libraries, and a Wasm runtime itself could be running as CLI in a terminal emulator. IMO seems premature to exclude WASI from that.

Reply:
@MaxDesiatov I want to avoid accidentally opting us into a state where the output is garbled. If we're not certain a feature works properly, it's better to disable it now and come back to it later than to preemptively enable it.

Pipes are problematic in this way because you can run a series of pipes that ultimately ends at a terminal emulator that supports ANSI escape codes, but actually determining whether outputtting ANSI escape codes is one of the Hard Problems™ in Computer Science™.

But, since @kateinoigakukun pointed out that `isatty()` is fully implemented, we can just defer to it. (Adding a flag to enable writing escape codes to an arbitrary file would be out of the scope of this PR, but we can chat about it in the forums.)